perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_tcsMailsTest.java

```diff
@@ -0,0 +1,195 @@
+import cz.metacentrum.perun.core.bl.AttributesManagerBl;
+import cz.metacentrum.perun.core.bl.ModulesUtilsBl;
+import cz.metacentrum.perun.core.bl.PerunBl;
+import cz.metacentrum.perun.core.bl.UsersManagerBl;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import org.hamcrest.CoreMatchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Michal Stava stavamichal@gmail.com
+ */
+public class urn_perun_user_attribute_def_virt_tcsMailsTest {
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_virt_tcsMailsTest.class);
+
+	private final urn_perun_user_attribute_def_virt_tcsMails classInstance = new urn_perun_user_attribute_def_virt_tcsMails();
+	private final AttributeDefinition tcsMailsAttrDef = classInstance.getAttributeDefinition();
+
+	private final User user = new User(10, "Joe", "Doe", "W.", "", "");
+
+	private final String email1 = "email1@mail.cz";
+	private final String email2 = "email2@mail.cz";
+	private final String email3 = "email3@mail.cz";
+	private final String email4 = "email4@mail.cz";
+	private final String email5 = "email5@mail.cz";
+
+	Attribute preferredMailAttr = setUpUserAttribute(1, "preferredMail", String.class.getName(), email1);
+	Attribute isMailAttr = setUpUserAttribute(2, "ISMail", String.class.getName(), email2);
+	Attribute publicMailsAttr = setUpUserAttribute(3, "publicAliasMails", ArrayList.class.getName(), new ArrayList(Arrays.asList(email3, email4)));
+	Attribute privateMailsAttr = setUpUserAttribute(4, "privateAliasMails", ArrayList.class.getName(), new ArrayList(Arrays.asList(email4, email5)));
+	Attribute o365MailsAttr = setUpUserAttribute(5, "o365EmailAddresses:mu", ArrayList.class.getName(), new ArrayList(Arrays.asList(email1, email3, email5)));
+
+	private PerunSessionImpl sess;
+
+	@Before
+	public void setUp() throws Exception {
+		tcsMailsAttrDef.setId(100);
+		//prepare mocks
+		sess = mock(PerunSessionImpl.class);
+		PerunBl perunBl = mock(PerunBl.class);
+		AttributesManagerBl am = mock(AttributesManagerBl.class);
+		UsersManagerBl um = mock(UsersManagerBl.class);
+		ModulesUtilsBl mu = mock(ModulesUtilsBl.class);
+		when(sess.getPerunBl()).thenReturn(perunBl);
+		when(perunBl.getAttributesManagerBl()).thenReturn(am);
+		when(perunBl.getUsersManagerBl()).thenReturn(um);
+		when(perunBl.getModulesUtilsBl()).thenReturn(mu);
+		when(mu.getUserFromMessage(eq(sess), any(String.class))).thenReturn(user);
```

Comment:
```
when(perunBl.getModulesUtilsBl()).thenReturn(mu);
when(mu.getUserFromMessage(eq(sess), any(String.class))).thenReturn(user);
```
You don't seem to be using above two methods, hence no need to mock them.

Reply:
Removed.